tests/dialects/test_tsql.py

```diff
@@ -1878,3 +1878,11 @@ def test_declare(self):
             "DECLARE vendor_cursor CURSOR FOR SELECT VendorID, Name FROM Purchasing.Vendor WHERE PreferredVendorStatus = 1 ORDER BY VendorID",
             check_command_warning=True,
         )
+
+    def test_scope_resolution_op(self):
+        self.validate_identity("SELECT ::FOO(a, b)")
+        self.validate_identity("SELECT bar::BAZ(1, 2)")
+        self.validate_identity("SELECT GEOGRAPHY::POINT([latitude], [longitude], 4326) FROM tbl")
```

Comment:
This is problematic after all, T-SQL is case-sensitive when it comes to these system functions:

```
1> SELECT GEOGRAPHY::POINT(34.2, 34.2, 4326)
2> go
Msg 6506, Level 16, State 10, Server af4359a4e415, Line 1
Could not find method 'POINT' for type 'Microsoft.SqlServer.Types.SqlGeography' in assembly 'Microsoft.SqlServer.Types'
1> SELECT GEOGRAPHY::Point(34.2, 34.2, 4326)
2> go
  
--
0x
```

Reply:
Maybe we can switch off the function name normalization only for anonymous functions in T-SQL and control this behavior with a generator flag?

Reply:
Or if we can prove this is the only case where case-sensitivity matters, check the parent of the `Func` and only normalize it if it's not a `ScopeResolution`.

Reply:
I tried a few different functions like date/time related ones and they all seem to be case insensitive except these, will disable normaliaztion for `ScopeResolution` funcs

Reply:
Cool, let's only do this for T-SQL

Reply:
It's awkward to _not_ normalize some of the functions in a dialect; For example, as I mentioned in T-SQL the builtin functions like `DATEDIFF`, `CONVERT` etc are case-insensitive in contrast to the CLR functions like `geography::Point` which are case-sensitive.

In order not to be intrusive to the rest of T-SQL, I chose to manually generate the function RHS of `exp.ScopeResolution`; If we want to normalize all functions in T-SQL I can follow up with a commit and change the PR to breaking.